File: app/src/main/java/com/donald/musictheoryapp/Music/ScoreView/SStaff.java
package com.donald.musictheoryapp.Music.ScoreView;

import android.graphics.Canvas;

// TODO: REMOVE STAFF HEIGHT
class SStaff extends ScoreStamp
{
	public static final int NO_OF_LINES = 5;

	private float width;

	protected SStaff(ScoreView view)
	{
		super(view);
		width = 0;
	}

	void setWidth(float width) { this.width = width; }

	protected float width() { return width; }

	@Override
	protected void onDraw(Canvas canvas, float posX, float posY)
	{
		for(int i = 0; i < NO_OF_LINES; i++)
		{
			canvas.drawLine(
				posX,
				posY + staffStepHeight() * 2 * i,
				posX + width,
				posY + staffStepHeight() * 2 * i,
				linePaint());
		}
	}
}

File: app/src/main/java/com/donald/musictheoryapp/music/scoreview/SStaff.kt
package com.donald.musictheoryapp.music.scoreview

import android.graphics.Canvas

// TODO: REMOVE STAFF HEIGHT
internal class SStaff(private val view: ScoreView) {

    var width = 0F

     fun onDraw(canvas: Canvas, posX: Float, posY: Float) {
        val staffStepHeight = REL_STAFF_STEP_HEIGHT * view.glyphSize
        val linePaint = view.linePaint
        for (i in 0 until NO_OF_LINES_IN_STAFF) {
            canvas.drawLine(
                posX,
                posY + staffStepHeight * 2 * i,
                posX + width,
                posY + staffStepHeight * 2 * i,
                linePaint
            )
        }
    }

}